quint/src/cliCommands.ts

```diff
@@ -369,7 +369,12 @@ export function runTests(prev: TypecheckedStage): CLIProcedure<TestedStage> {
     } // else: we have handled the case of module not found already
 
     const errors = namedErrors.map(([_, e]) => e)
-    return right({ ...testing, passed, failed, ignored, errors })
+    const result = { ...testing, passed, failed, ignored, errors }
+    if (errors.length == 0 && failed.length == 0) {
+      return right(result)
+    } else {
+      return left({msg: "Tests failed", stage: { ...result, errors }})
```

Comment:
`errors` is already inside `result`, so I think we can do this:
```suggestion
      return left({msg: "Tests failed", stage: { ...result }})
```